packages/strapi-utils/lib/buildQuery.js

```diff
@@ -105,7 +105,9 @@ const buildQuery = ({ model, filters = {}, ...rest }) => {
           field,
         });
 
-        const { type } = _.get(assocModel, ['attributes', attribute], {});
+        const { type } = (operator === 'null')
```

Comment:
no need to change this :)

Reply:
call castValueToType with the operator and add the operator check there directly ;)